framework/src/modules/chain/logic/init_transaction.js

```diff
@@ -163,43 +48,15 @@ class Transaction {
 
 	// TODO: remove after https://github.com/LiskHQ/lisk/issues/2424
 	dbRead(raw) {
-		if (!raw.t_id) {
+		if (!raw.t_id || !raw.t_type) {
 			return null;
 		}
 
-		const transactionJSON = {
-			id: raw.t_id,
-			height: raw.b_height,
-			blockId: raw.b_id || raw.t_blockId,
-			type: parseInt(raw.t_type),
-			timestamp: parseInt(raw.t_timestamp),
-			senderPublicKey: raw.t_senderPublicKey,
-			requesterPublicKey: raw.t_requesterPublicKey,
-			senderId: raw.t_senderId,
-			recipientId: raw.t_recipientId,
-			recipientPublicKey: raw.m_recipientPublicKey || null,
-			amount: raw.t_amount,
-			fee: raw.t_fee,
-			signature: raw.t_signature,
-			signSignature: raw.t_signSignature,
-			signatures: raw.t_signatures ? raw.t_signatures.split(',') : [],
-			confirmations: parseInt(raw.confirmations || 0),
-			asset: {},
-		};
-
-		const assetDbRead = assetDbReadMap.get(transactionJSON.type);
-
-		if (!assetDbRead) {
-			throw new Error(`Unknown transaction type ${transactionJSON.type}`);
-		}
-
-		const asset = assetDbRead(raw);
-
-		if (asset) {
-			transactionJSON.asset = Object.assign(transactionJSON.asset, asset);
-		}
+		const transactionJSON = this.transactionClassMap(
+			parseInt(raw.t_type)
+		).dbRead(raw);
 
-		return this.fromJson(_.omitBy(transactionJSON, _.isNull));
+		return this.fromJson(transactionJSON);
```

Comment:
why are we omitting the cleanup of `nulls` now?

Reply:
this logic was abstracted to `baseTransaction.dbRead`